packages/lesswrong/components/hooks/useBookmark.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, MouseEvent, useEffect, useCallback } from "react";
+import { gql, useMutation } from "@apollo/client";
+import { useCurrentUser } from "@/components/common/withUser";
+import { useDialog } from "@/components/common/withDialog";
+import { useTracking } from "@/lib/analyticsEvents";
+import type { ForumIconName } from "@/components/common/ForumIcon";
+import { Components } from "@/lib/vulcan-lib/components";
+import { fragmentTextForQuery } from "@/lib/vulcan-lib/fragments";
+import { useMulti } from "@/lib/crud/withMulti";
+
+export interface UseBookmarkResult {
+  isBookmarked: boolean;
+  toggleBookmark: (event?: MouseEvent) => void;
+  loading: boolean;
+  icon: ForumIconName;
+  labelText: string;
+  hoverText: string;
+}
+
+const TOGGLE_BOOKMARK_MUTATION = gql`
+  mutation ToggleBookmarkMutation($input: ToggleBookmarkInput!) {
+    toggleBookmark(input: $input) {
+      data {
+        ...BookmarksDefaultFragment
+      }
+    }
+  }
+  ${fragmentTextForQuery('BookmarksDefaultFragment')}
+`;
+
+export const useBookmark = (
+  documentId: string | undefined,
```

Comment:
Is there any time you're ever going to be calling this with an `undefined` documentId?